app/src/main/java/com/ikarabulut/GameSettings.java

```diff
@@ -32,37 +37,135 @@ public void getAllSettings() {
 
     public void setEvolutions() {
         display.evolutionsPrompt();
-        evolutions = Integer.parseInt(input.nextLine());
+        int retries = 3;
+
+        while (retries > 0) {
+            try {
+                evolutions = Integer.parseInt(input.nextLine());
+                return;
+            } catch (Exception e) {
+                System.err.print("An error occurred, please enter an integer. Try again: ");
+                retries = retries - 1;
+            }
+            if (retries == 0) {
+                System.err.print("You have failed to choose a valid number of evolutions. Game exiting");
+                System.exit(1);
+            }
+        }
     }
 
+
     public void setRows() {
         display.rowsPrompt();
-        rows = Integer.parseInt(input.nextLine());
+        int intSelection = 0;
+        int retries = 3;
+
+        while (retries > 0) {
```

Comment:
Now that you have a pattern going for how you're accepting user input, you probably noticed that `setRows`, `setColumns`, `setAliveSymbol`, etc all have very similar logic. This is the point where I'd at least start thinking "is there a way I could extract this out into its own method?". Not that you have to do that, just something to think about.

Reply:
A good place to review and apply the Flocking Rules from 99 Bottles. 👍